### ISSUE TITLE
Assignment 3 Query

### ISSUE THREAD

**Issue by kumar-srikanth** [vincenzocoia/BAIT509#23]
Hi Vincenzo

In Exercise 4, you have mentioned not to take mpg and name as predictors. So should we use cost as a SVM vector to predict mileage (0,1) since the ISLR mentions in their question to use cost to predict mileage.

Thanks

**Comment by fish117** [comment]
Hi Vincenzo
In Exercise 4, you said we should report the validation error in all cases. So should we use the accuracy score of the validation dataset? Or should we get the prediction value then calculate the MSE?
Thanks.
Iris Xu

**Comment by kumar-srikanth** [comment]
Hi Vincenzo
In Exercise 4, when we are fitting the two models of SVC, should we just change the "C" hyper parameter or any other as well? Changing other parameters does not give any noticeable change in the accuracy.

Srikanth

**Comment by vincenzocoia** [comment]
> In Exercise 4, you have mentioned not to take mpg and name as predictors. So should we use cost as a SVM vector to predict mileage (0,1) since the ISLR mentions in their question to use cost to predict mileage.

Use all columns that aren't `mpg` and `name` as predictors for the SVM.

> In Exercise 4, you said we should report the validation error in all cases. So should we use the accuracy score of the validation dataset? Or should we get the prediction value then calculate the MSE?

Yes, use the accuracy score (or error=1-accuracy) by predicting on the validation set, and comparing to the true response on the validation set.

> In Exercise 4, when we are fitting the two models of SVC, should we just change the "C" hyper parameter or any other as well? Changing other parameters does not give any noticeable change in the accuracy.

You should also change the kernel-specific hyperparameter -- that is, "gamma" for the radial kernel. If there is no noticeable change, that's OK. You're not required to do a thorough investigation as to optimal values of the tuning parameters -- just explore a small grid.

**Comment by Karansheraz** [comment]
@vincenzocoia can we use crossValidation and fit the model to a parameter grid to get the best fit.
Use test scores of training set by CV to rank the best parameter set?
http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV

**Comment by vincenzocoia** [comment]
Great find, @Karansheraz ! That would actually be better than what the assignment is asking. Do use it if you get the chance.
